src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3574,68 +3574,168 @@ internal Binder GetBinder(CSharpSyntaxNode syntaxNode)
+                {
+                    diagnostics.Add(ErrorCode.ERR_PartialPropertyDuplicateImplementation, currentProperty.GetFirstLocation());
+                }
+                else if (currentProperty.IsPartialDefinition &&
+                            (prevProperty.IsPartialDefinition || (prevProperty.OtherPartOfPartial is SourcePropertySymbol otherDefinition && (object)otherDefinition != currentProperty)))
+                {
+                    diagnostics.Add(ErrorCode.ERR_PartialPropertyDuplicateDefinition, currentProperty.GetFirstLocation());
+                }
+                else
                 {
-                    // partial implementations not paired with a definition
-                    if (method.IsPartialImplementation && method.OtherPartOfPartial is null)
+                    var (currentGet, prevGet) = ((SourcePropertyAccessorSymbol?)currentProperty.GetMethod, (SourcePropertyAccessorSymbol?)prevProperty.GetMethod);
+                    if (currentGet != null || prevGet != null)
                     {
-                        diagnostics.Add(ErrorCode.ERR_PartialMethodMustHaveLatent, method.GetFirstLocation(), method);
+                        var accessorName = (currentGet ?? prevGet)!.Name.AsMemory();
+                        mergeAccessors(ref membersByName, accessorName, currentGet, prevGet);
                     }
-                    else if (method is { IsPartialDefinition: true, OtherPartOfPartial: null, HasExplicitAccessModifier: true })
+
+                    var (currentSet, prevSet) = ((SourcePropertyAccessorSymbol?)currentProperty.SetMethod, (SourcePropertyAccessorSymbol?)prevProperty.SetMethod);
+                    if (currentSet != null || prevSet != null)
                     {
-                        diagnostics.Add(ErrorCode.ERR_PartialMethodWithAccessibilityModsMustHaveImplementation, method.GetFirstLocation(), method);
+                        var accessorName = (currentSet ?? prevSet)!.Name.AsMemory();
+                        mergeAccessors(ref membersByName, accessorName, currentSet, prevSet);
                     }
+
+                    if ((object)membersByName == _lazyEarlyAttributeDecodingMembersDictionary)
+                    {
+                        // Avoid mutating the cached dictionary and especially avoid doing this possibly on multiple threads in parallel.
+                        membersByName = new Dictionary<ReadOnlyMemory<char>, ImmutableArray<Symbol>>(membersByName, ReadOnlyMemoryOfCharComparer.Instance);
+                    }
+
+                    membersByName[name] = FixPartialMember(membersByName[name], prevProperty, currentProperty);
                 }
-            }
 
-            memberNames.Free();
+                void mergeAccessors(ref Dictionary<ReadOnlyMemory<char>, ImmutableArray<Symbol>> membersByName, ReadOnlyMemory<char> name, SourcePropertyAccessorSymbol? currentAccessor, SourcePropertyAccessorSymbol? prevAccessor)
```

Comment:
>mergeAccessors [](http://example.com/codeflow?start=21&length=14)

We generally try to avoid nested local functions, although I understand that means a fairly long list of parameters in this case :-/

Reply:
I am satisfied with the way this is factored currently. Let me know if you feel strongly about it and we can adjust.